rllib/algorithms/sac/sac_learner.py

```diff
@@ -0,0 +1,117 @@
+import abc
+import numpy as np
+
+from typing import Any, Dict
+
+from ray.rllib.algorithms.sac.sac import SACConfig
+from ray.rllib.core.learner.learner import Learner
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.lambda_defaultdict import LambdaDefaultDict
+from ray.rllib.utils.metrics import LAST_TARGET_UPDATE_TS, NUM_TARGET_UPDATES
+from ray.rllib.utils.schedules.scheduler import Scheduler
+from ray.rllib.utils.typing import ModuleID
+
+# Now, this is double defined: In `SACRLModule` and here. I would keep it here
+# or push it into the `Learner` as these are recurring keys in RL.
+QF_PREDS = "qf_preds"
+QF_TARGET_PREDS = "qf_target_preds"
+QF_TWIN_PREDS = "qf_twin_preds"
+
+
+# TODO (simon): Add and remove variables, if module is added or removed.
+class SACLearner(Learner):
+    @override(Learner)
+    def build(self) -> None:
+        # Store the current alpha in log form. We need it during optimization
+        # in log form.
+        self.curr_log_alpha: Dict[ModuleID, Scheduler] = LambdaDefaultDict(
```

Comment:
nit: This won't be a `Scheduler` here, correct? "torch.Tensor" ?

Reply:
nit: can we rename to `self.log_alpha_per_module`

Reply:
No, as far as I know a `Scheduler` makes the update by timestep while this one does it via the optimizer. The type is going to be a `nn.Parameter` to be precise (wrapping the `torch.Tensor`